tests/unit/utils/Utils.h

```diff
@@ -55,4 +58,44 @@ SGStringList<T> generateRandomStringData(
 	return strings;
 }
 
+struct NeuralLayerTestUtil
+{
+	template <typename T>
+	static SGMatrix<T> create_rand_sgmat(
+	    int32_t num_rows, int32_t num_cols, T lower_bound, T upper_bound)
+	{
+		auto data_batch = SGMatrix<T>(num_rows, num_cols);
+		for (int32_t i = 0; i < data_batch.num_rows * data_batch.num_cols; i++)
+		{
+			data_batch[i] = CMath::random(lower_bound, upper_bound);
```

Comment:
actually, there should also be a more efficient way to general an array of random normals. Check the array filling procedures in `Random.h`